src/actions/semrush/semrush-options-action.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Country code option action for SEMrush.
+ *
+ * @package Yoast\WP\SEO\Actions\SEMrush
+ */
+
+namespace Yoast\WP\SEO\Actions\SEMrush;
+
+use \Yoast\WP\SEO\Helpers\Options_Helper;
+
+/**
+ * Class SEMrush_Options_Action
+ */
+class SEMrush_Options_Action {
+
+	/**
+	 * @var Options_Helper
+	 */
+	protected $options_helper;
+
+	/**
+	 * SEMrush_Options_Action constructor.
+	 *
+	 * @param \Yoast\WP\SEO\Helpers\Options_Helper $options_helper The WPSEO options helper.
+	 */
+	public function __construct( Options_Helper $options_helper ) {
+		$this->options_helper = $options_helper;
+	}
+
+	/**
+	 * Stores SEMrush country code in the WPSEO options.
+	 *
+	 * @param string $country_code The country code to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function set_country_code( $country_code ) {
+		// The country code has already been validated at this point. No need to do that again.
+		$success = $this->options_helper->set( 'semrush_country_code', $country_code );
+
+		if ( $success ) {
+			return (object) [
+				'success' => true,
+				'status'  => 200,
+			];
+		}
+		else {
+			return (object) [
+				'success' => false,
+				'status'  => 500,
+			];
+		}
```

Comment:
I have two things regarding this bit of code.

First: The `else` is unnecessary. Just returning the 'error object' immediately should be fine.
Secondly: Returning a `success: false` to the browser doesn't really give any context as to why the saving failed. I'd suggest adding a proper error message as well, similar to how `src/exceptions/oauth/oauth-authentication-failed-exception.php` does that.

Reply:
Fixed.